base/src/main/java/uk/ac/ebi/atlas/utils/ReactomeClient.java

```diff
@@ -1,39 +1,108 @@
+import com.google.gson.JsonSyntaxException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.web.client.RestClientException;
 import org.springframework.web.client.RestTemplate;
 
 import javax.inject.Inject;
 import javax.inject.Named;
-import java.text.MessageFormat;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 @Named
 public class ReactomeClient {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ReactomeClient.class);
 
+    static final String URL = "https://reactome.org/ContentService/data/query/ids";
+    static final int QUERY_MAX_SIZE = 20;  // https://reactome.org/ContentService/#!/query/findByIdsUsingPOST
+    private static final Gson GSON = new Gson();
+    private static final String STATIC_ID_FIELD = "stId";
+    private static final String DISPLAY_NAME_FIELD = "displayName";
+
     private final RestTemplate restTemplate;
 
     @Inject
     public ReactomeClient(RestTemplate restTemplate) {
         this.restTemplate = restTemplate;
     }
 
-    public Optional<String> fetchPathwayNameFailSafe(String reactomeId) {
-        String reactomeURL = "https://reactome.org/ContentService/data/query/{0}/displayName";
-        String url = MessageFormat.format(reactomeURL, reactomeId);
+    public Optional<String> getPathwayName(String reactomeId) {
+        return Optional.ofNullable(getPathwayNames(ImmutableList.of(reactomeId)).get(reactomeId));
+    }
+
+    public ImmutableMap<String, String> getPathwayNames(Collection<String> stableIds) {
+        List<List<String>> partitions = Lists.partition(ImmutableList.copyOf(stableIds), QUERY_MAX_SIZE);
+
+        ImmutableMap.Builder<String, String> mappedStableIdsBuilder = ImmutableMap.builder();
+        for (List<String> partition : partitions) {
+            mappedStableIdsBuilder.putAll(fetchPathwayNames(partition));
+        }
+
+        ImmutableMap<String, String> mappedStableIds = mappedStableIdsBuilder.build();
+        logMissingIds(stableIds, mappedStableIds);
+
+        return mappedStableIds;
+    }
+
+    private ImmutableMap<String, String> fetchPathwayNames(Collection<String> stableIds) {
+        String postData = stableIds.stream().collect(Collectors.joining(","));
 
         try {
-            return Optional.of(StringUtils.trim(restTemplate.getForObject(url, String.class)));
+            return parseJsonResponse(restTemplate.postForObject(URL, postData, String.class), stableIds);
+        } catch (JsonSyntaxException e) {
+            LOGGER.error("Invalid JSON returned from Reactome API");
+            return ImmutableMap.of();
         } catch (RestClientException e) {
-            LOGGER.warn("Reactome ID \"{}\" could not be found", reactomeId);
-            return Optional.empty();
+            LOGGER.error("There was an error retrieving pathway names from Reactome");
```

Comment:
Use the error logs to tell the future you what the exception was, they'll be grateful.